### PR TITLE
Upstream `spacelift`

### PR DESCRIPTION
**Pull request by milldr**
## what
- Upstreaming the `spacelift` component

## why
- This update sets up catalog support for the `spacelift` admin stacks
- Allows multiple admin stacks

## references
- https://github.com/cloudposse/infra-live

### CODE REVIEW

File: modules/s3-bucket/main.tf [hunk 1]
```diff
@@ -1,53 +1,83 @@
...
+module "bucket_policy" {
+  source  = "cloudposse/iam-policy/aws"
+  version = "0.3.0"
+
+  iam_policy_statements = var.source_policy_documents
+
+  context = module.introspection.context
+}
+
 module "s3_bucket" {
```
Comment (Gowiem) [inline modules/s3-bucket/main.tf:33]: @milldr did you mean to include the S3 bucket changes into this PR?

Comment (milldr) [reply modules/s3-bucket/main.tf]: I did not! Updating this PR today and will hopefully get this pushed through

**Comment by milldr** [comment]
upstream required by [CPLIVE-303](https://cloudposse.atlassian.net/browse/CPLIVE-303)

File: modules/spacelift/variables.tf [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+
+variable "runner_image" {
+  type        = any
+  description = "Full address & tag of the Spacelift runner image (e.g. on ECR)"
 }
 
 variable "worker_pool_id" {
-  type        = string
-  description = "Alpha-numberic worker poold ID of the Spacelift worker pool to use"
+  type        = any
```
Comment (nitrocode) [inline modules/spacelift/variables.tf:12]: Let's retain the original types

We had to switch to `any` in a previous client but no longer need to

File: modules/spacelift/bin/spacelift-write-vars [hunk 3]
```diff
@@ -1,20 +0,0 @@
...
-  fi
-
-  echo "Writing Stack variables to spacelift.auto.tfvars.json for Spacelift..."
-
-  atmos --config-dir ../../../stacks --terraform-dir ../ terraform write varfile "$ATMOS_COMPONENT" --stack="$ATMOS_STACK" >/dev/null
-  mv -f "${ATMOS_STACK}-${ATMOS_COMPONENT}.terraform.tfvars.json" spacelift.auto.tfvars.json
-  jq . <spacelift.auto.tfvars.json
-}
-
-main
```
Comment (nitrocode) [inline modules/spacelift/bin/spacelift-write-vars:20]: Is this no longer needed ? I think we may still be using this in our newest client

Comment (milldr) [reply modules/spacelift/bin/spacelift-write-vars]: reverted

File: modules/spacelift/variables.tf [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
 
 variable "worker_pool_id" {
   type        = string
-  description = "Alpha-numberic worker poold ID of the Spacelift worker pool to use"
+  description = "DEPRECATED: Use worker_pool_name_id_map instead. Worker pool ID"
+  default     = ""
+}
+
+variable "worker_pool_name_id_map" {
+  type        = any
```
Comment (nitrocode) [inline modules/spacelift/variables.tf:18]: map(any) ?

File: modules/spacelift/variables.tf [hunk 5]
```diff
@@ -1,11 +1,23 @@
-variable "runner_image" {
+variable "region" {
   type        = string
-  description = "Full address & tag of the Spacelift runner image (e.g. on ecr)"
+  description = "AWS Region"
+}
+
+variable "runner_image" {
+  type        = any
```
Comment (nitrocode) [inline modules/spacelift/variables.tf:7]: string ?

File: modules/spacelift/variables.tf [hunk 6]
```diff
@@ -41,3 +64,134 @@ variable "terraform_version_map" {
...
 }
+
+variable "stack_config_path_template" {
+  type        = string
+  description = "Stack config path template"
+  default     = "stacks/%s.yaml"
+}
+
+variable "policies_available" {
+  type        = any
```
Comment (nitrocode) [inline modules/spacelift/variables.tf:75]: list(string)

File: modules/spacelift/variables.tf [hunk 7]
```diff
@@ -41,3 +64,134 @@ variable "terraform_version_map" {
...
+    "git_push.proposed-run",
+    "git_push.tracked-run",
+    "plan.default",
+    "trigger.dependencies",
+    "trigger.retries"
+  ]
+}
+
+variable "policies_enabled" {
+  type        = any
```
Comment (nitrocode) [inline modules/spacelift/variables.tf:87]: please change all any's to their respective type.

we had a customer complaint about this recently.

File: modules/spacelift/variables.tf (hunk 4 above)
Comment (milldr) [reply modules/spacelift/variables.tf:18]: ah misunderstood previous request. Updating all variables now

**Review by nitrocode — APPROVED** [review]

**Pull request merged**